File: Dongyeon/2193.py
N = int(input())

def fib(n, d):
    if n in d.keys():
        return d[n]
    else:
        d[n] = fib(n-1, d) + fib(n-2, d)
        return d[n]

d = {1 : 1,
     2 : 1,
     3 : 2}

print(fib(N, d))

File: wi-seong-cheol/BOJ/0x10 DP/2193.cpp
//
//  2193.cpp
//  wi-seong-cheol
//
//  Created by wi_seong on 2022/12/02.
//

/*
 [Input]
 3
 [Output]
 2
 */
// 시간복잡도: O(n)
// 최악시간: 90
// 난이도: Silver 3
// Timer: 5m
// Url: https://www.acmicpc.net/problem/2193

#include <iostream>

using namespace std;
#define ll long long
int n;
ll d[91];

int main() {
    ios::sync_with_stdio(0);
    cin.tie(0);

    cin >> n;
    d[1] = 1;
    d[2] = 1;
    for(int i = 3; i <= n; i++)
        d[i] = d[i - 1] + d[i - 2];
    cout << d[n];
    return 0;
}
